### PR TITLE
Added support for uploading files via form data

### PR DESCRIPTION
**Pull request by shreys7**
Codegens needed to be changed:
- golang
- http
- js-fetch
- js-jquery
- js-xhr
- nodejs-native
- ocaml
- php-pecl-http
- powershell-restmethod
- python-http.client
- python-requests
- ruby
- shell-httpie
- swift

### CODE REVIEW

File: codegens/http/lib/util.js [hunk 1]
```diff
@@ -144,9 +217,19 @@ function getBody (request, trimRequestBody) {
...
               requestBody += 'Content-Disposition: form-data; name="';
-              requestBody += `${(trimRequestBody ? property.key.trim() : property.key)}"; filename="<FILE_NAME >"`;
-              requestBody += `\n${(trimRequestBody ? property.value.trim() : property.value)}\n`;
+              requestBody += `${(trimRequestBody ? property.key.trim() : property.key)}"; filename="`;
+              requestBody += `${fileName}"\n`;
+              if (contentTypeHeaderMap[fileExtension]) {
+                requestBody += `Content-Type: ${contentTypeHeaderMap[fileExtension]}\n\n`;
+              }
+              else {
+                requestBody += 'Content-Type: content-type header here\n\n';
```
Comment (abhijitkane) [inline codegens/http/lib/util.js]: maybe something more placeholder-y, like `<Content-Type header here>`

File: codegens/js-fetch/lib/index.js [hunk 2]
```diff
@@ -42,7 +42,9 @@ function parseFormData (body, trim) {
   _.forEach(body, function (data) {
     if (!data.disabled) {
       if (data.type === 'file') {
-        bodySnippet += `formdata.append("${sanitize(data.key, trim)}", "${sanitize(data.src, trim)}");\n`;
+        bodySnippet += `// if access to fs, use fs.createReadStream('${data.src}') ` +
```
Comment (abhijitkane) [inline codegens/js-fetch/lib/index.js]: too complicated. We can move this out as an option later. For now, let's assume it's a browser environment and there's no `fs`.
Also, in the `formdata.append` line, there's no space after the first comma, but there is one after the second.

File: codegens/js-fetch/lib/index.js [hunk 3]
```diff
@@ -185,7 +187,7 @@ function convert (request, options, callback) {
   var indent = options.indentType === 'Tab' ? '\t' : ' ',
     trim = options.trimRequestBody,
     headers, body,
-    codeSnippet = '',
+    codeSnippet = 'var fs = require(\'fs\');\n',
```
Comment (abhijitkane) [inline codegens/js-fetch/lib/index.js]: Can remove. Can come in as an option later.

File: codegens/http/lib/util.js [hunk 4]
```diff
@@ -144,9 +217,19 @@ function getBody (request, trimRequestBody) {
               requestBody += `\n${(trimRequestBody ? property.value.trim() : property.value)}\n`;
             }
             else if (property.type === 'file') {
+              var pathArray = property.src.split('/'),
+                fileName = pathArray[pathArray.length - 1],
```
[the remark this answers is not in the document]
Comment (shreys7) [reply codegens/http/lib/util.js]: using path.sep now

File: codegens/js-xhr/lib/index.js [hunk 5]
```diff
@@ -44,8 +44,9 @@ function parseFormData (body, trim) {
       /* istanbul ignore next */
       /* ignoring because the file src is not stored in postman collection" */
       if (data.type === 'file') {
-        bodySnippet += `data.append("${sanitize(data.key, trim)}", "${sanitize(data.src, trim)}", `;
-        bodySnippet += `"${sanitize(data.key, trim)}");\n`;
+        var pathArray = data.src.split('/'),
+          fileName = pathArray[pathArray.length - 1];
+        bodySnippet += `data.append("${sanitize(data.key, trim)}",fileInput.files[0], "${fileName}");\n `;
```
[the remark this answers is not in the document]
Comment (shreys7) [reply codegens/js-xhr/lib/index.js]: done

File: codegens/nodejs-native/lib/parseRequest.js [hunk 6]
```diff
@@ -36,12 +41,14 @@ function generateMultipartFormData (requestbody) {
         const key = dataArrayElement.key.replace(/"/g, '\'');
 
         if (dataArrayElement.type === 'file') {
-          const filename = 'filename=\\"{Insert_File_Name}\\"',
+          var pathArray = dataArrayElement.src.split('/'),
```
[the remark this answers is not in the document]
Comment (shreys7) [reply codegens/nodejs-native/lib/parseRequest.js]: using path.sep now

File: codegens/php-pecl-http/lib/util/parseBody.js [hunk 7]
```diff
@@ -44,7 +44,7 @@ module.exports = function (request, indentation, bodyTrim) {
           });
           bodyFileMap = _.map(_.filter(enabledBodyList, {'type': 'file'}), function (value) {
             return (`${indentation.repeat(2)}array('name' => '${sanitize(value.key, bodyTrim)}', ` +
-                            `'type' => '${sanitize(value.type, bodyTrim)}', ` +
+                            '\'type\' => \'content-type header\', ' +
```
[the remark this answers is not in the document]
Comment (shreys7) [reply codegens/php-pecl-http/lib/util/parseBody.js]: done

File: codegens/powershell-restmethod/lib/index.js [hunk 8]
```diff
@@ -39,12 +39,26 @@ function parseFormData (body, trim) {
   var bodySnippet = '$multipartContent = [System.Net.Http.MultipartFormDataContent]::new()\n';
   _.forEach(body, function (data) {
     if (!data.disabled) {
-      bodySnippet += '$stringHeader = ' +
+      if (data.type === 'text') {
+        bodySnippet += '$stringHeader = ' +
                             '[System.Net.Http.Headers.ContentDispositionHeaderValue]::new("form-data")\n' +
```
[the remark this answers is not in the document]
Comment (shreys7) [reply codegens/powershell-restmethod/lib/index.js]: done

File: codegens/shell-httpie/lib/util/helpers.js [hunk 9]
```diff
@@ -155,10 +155,13 @@ module.exports = {
...
-            if (typeof param.value === 'string') {
-              return ' ' + Sanitize.quote(param.key) + '=' + Sanitize.quote(param.value);
+            if (param.type === 'text') {
+              if (typeof param.value === 'string') {
+                return ' ' + Sanitize.quote(param.key) + '=' + Sanitize.quote(param.value);
+              }
+              return ' ' + param.key + ':=' + param.value;
             }
-            return ' ' + param.key + ':=' + param.value;
+            return ' ' + Sanitize.quote(param.key) + '@' + param.src;
```
[the remark this answers is not in the document]
Comment (shreys7) [reply codegens/shell-httpie/lib/util/helpers.js]: I searched on the docs and tried manually, apparently we cannot add spaces between form data field name and the file path

**Pull request merged**